skbeam/core/accumulators/binned_statistic.py

```diff
@@ -130,7 +136,13 @@ def __init__(self, sample, statistic='mean',
         # Compute the bin number each sample falls into.
         Ncount = {}
         for i in np.arange(self.D):
-            Ncount[i] = np.digitize(sample[:, i], self.edges[i])
+            # Apply mask in a non-ideal way by setting value outside range.
+            # Would be better to do this using bincount "weights", perhaps.
+            thissample = sample[:, i]
+            thismask = mask[:, i]
+            thissample[thismask == 0] = (self.edges[i][0] -
+                                         0.01 * (1+np.fabs(self.edges[i][0])))
+            Ncount[i] = np.digitize(thissample, self.edges[i])
```

Comment:
Hi Julien,

Sure, I have added this to the constructor documentation.  One thing with this code is that (at least for LCLS usage patterns) the np.digitize only happens once when the RadialBinnedStatistic is setup.  Then the many images are projected (at 120Hz) without ever calling np.digitize again.  So for LCLS usage patterns the CPU time of the setup in the constructor is negligible compared to the 120Hz analysis of images.